client/src/pages/CalendarPage.js

```diff
@@ -108,25 +121,104 @@ const useStyles = makeStyles((theme) => ({
 export default function CalendarPage() {
     const classes = useStyles();
     const [minDate, setMinDate] = useState(new Date());
-    // may need fetch from server with useEffect hook
-    const [times, setTimes] = useState([
-        '16:30',
-        '17:00',
-        '17:30',
-        '18:00',
-        '18:30',
-        '19:00',
-        '19:30',
-        '20:00',
-        '20:30',
-        '21:00',
-        '21:30'
-    ])
-    const listTime = times.map((time) =>
-        <Button variant="outlined" fullWidth className={classes.timeBtn}>
-            <FiberManualRecordIcon className={classes.dotIcon} />&nbsp;&nbsp;{time}
-        </Button>
-    )
+    const [currentMonth, setCurrentMonth] = useState(''); //date of the first day of the month
+    const [times, setTimes] = useState([]);
+    const [timePeriods, setTimePeriods] = useState([]);
+    const [selectedDay, setSelectedDay] = useState(new Date());
+    const [selectedTime, setSelectedTime] = useState('');
+
+    const handleClickDay = (day) => {
+        if (selectedDay !== "" && new Date(day).getMonth() !== new Date(selectedDay).getMonth()) {
+            let yearMonth = new Date(day).toISOString();
+            setCurrentMonth(yearMonth);
+        };
+        setSelectedDay(day)
+    }
+
+    const handleClickTime = (e) => {
+        setSelectedTime(e.target.id.replace(/-/, ":"))
+    }
+
+    const TimeSlots = (start, end) => {
+        var start = moment(start, "HH:mm");
+        var end = moment(end, "HH:mm");
+        let times = [];
+        while (start < end) {
+            times.push(start.format("HH:mm"));
+            start.add(30, 'minutes')
+        };
+        setTimes(times)
+    };
+
+    const user = useContext(UserContext);
+    useEffect(() => {
+        let userId = user.userId;
```

Comment:
Use `const` for the `userId`.
Check out [this article](https://hacks.mozilla.org/2015/07/es6-in-depth-let-and-const/) to learn more about the differences between `var`, `let`, and `const`.